lm75bd/lm75bd.c

```diff
@@ -26,8 +27,31 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 }
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
-  
+  error_code_t errCode;
+
+  /* Selecting temperature register */
+  uint8_t sendData = 0;
+  errCode = i2cSendTo(devAddr, &sendData, 1);
+  if (errCode != ERR_CODE_SUCCESS) return errCode;
+
+  /* Receive temp reading */
+  uint8_t buff[2] = {0, 0};
+  errCode = i2cReceiveFrom(devAddr, buff, 2);
+  if (errCode != ERR_CODE_SUCCESS) return errCode;
+
+  /* Convert to celceus */
+  uint16_t readValue = 0;
+  readValue = ((uint16_t) buff[0] << 8) | buff[1];
+
+  /* Positive if 1st MSB is 0 */
+  if(buff[0] < 0b10000000){
+    readValue = readValue >> 5;
+    *temp = ((float) readValue) * 0.125;
+  } else {
+    readValue = ~readValue + 1;
+    readValue = readValue >> 5;
+    *temp =  ((float) readValue) * -0.125;
+  }
```

Comment:
This can be simplified to:
```suggestion
  int16_t regVal = (buff[0] << 8) | buff[1]; 
  *temp = (float)(regVal >> 5) * 0.125f;
```

This is because the register data is already in 2s complement format.

Reply:
ah yes, that's a better way of doing that!